source/main.lua

```diff
@@ -1,38 +1,44 @@
+import "CoreLibs/object"
+import "CoreLibs/graphics"
+import "CoreLibs/sprites"
+import "CoreLibs/timer"
 import "CoreLibs/crank"
 import "CoreLibs/qrcode"
-import "CoreLibs/timer"
 
-isNormalMode = false
+import "GameData"
+import "NormalScreen"
+import "QrCodeScreen"
 
-local menu = playdate.getSystemMenu()
+local gfx <const> = playdate.graphics
 
-local menuItem, error = menu:addMenuItem("normal", function()
-	isNormalMode = true
-end)
+local gameData = GameData()
+local normalScreen = NormalScreen(gameData)
+local qrCodeScreen = QrCodeScreen(gameData)
 
-local menuItem, error = menu:addMenuItem("code", function()
-	isNormalMode = false
-end)
+function init()
+	local menu = playdate.getSystemMenu()
+	menu:addMenuItem("normal", function()
+		qrCodeScreen:clear()
+		normalScreen:render()
+	end)
+	menu:addMenuItem("code", function()
+		normalScreen:clear()
+		qrCodeScreen:render()
+	end)
+	normalScreen:render()
+end
 
-ticks = 0
+function playdate.cranked()
+	local newTicks = playdate.getCrankTicks(1) or 0
+	if newTicks == 0 then return end
```

Comment:
It took me a bit to figure out the difference between this and what I last saw, but this is a nice change to simplify the `if ticks + newTicks== originalTIcks` logic that we had.